libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -1301,6 +1309,7 @@ int HTTPClient::handleHeaderResponse()
                     _canReuse = (headerLine[sizeof "HTTP/1." - 1] != '0');
                 }
                 _returnCode = headerLine.substring(9, headerLine.indexOf(' ', 9)).toInt();
+                _canReuse = (_returnCode != '0');
```

Comment:
Few problems here.

_canReuse sholuld only be cleared, not set by this (i.e. if its already false due to HTTP 1.0).

_returnCode is already an integer, not a char.

Also, *maybe* _returnCode !=0 and _returnCode < 500 (i.e. does a server error mean no connection reuse?  I don't know the answer yet.)

I think that results in:  `_canReuse = _canReuse && (_returnCode != 0)`

Reply:
applied in conflict resolution